frontend/packages/signless-transactions/src/hooks/use-create-session.ts

```
@@ -76,16 +71,58 @@ function useCreateSession(programId: HexString, metadata: ProgramMetadata | unde
 
     return api.voucher.update(session.key, voucher.id, { prolongDuration, balanceTopUp });
   };
+  type Options = Partial<{
+    onSuccess: () => void;
+    onError: (error: string) => void;
+    onFinally: () => void;
+    pair?: KeyringPair;
+  }>;
 
   const createSession = async (
     session: Session,
     voucherValue: number,
-    { shouldIssueVoucher, ...options }: Options & { shouldIssueVoucher: boolean },
+    pairToSave: KeyringPair | null,
+    { shouldIssueVoucher, vId, ...options }: Options & { shouldIssueVoucher: boolean } & { vId: string | void },
   ) => {
     if (!isApiReady) throw new Error('API is not initialized');
     if (!account) throw new Error('Account not found');
     if (!metadata) throw new Error('Metadata not found');
 
+    if (vId && pairToSave) {
+      const { signer } = await web3FromSource(account.meta.source);
+      const { signRaw } = signer;
+
+      if (!signRaw) {
+        throw new Error('signRaw is not a function');
+      }
+
+      const key = decodeAddress(pairToSave.address);
+      const duration = session.duration;
+      const allowedActions = session.allowedActions;
+
+      const payloadToSign = { key, duration, allowedActions };
+
+      if (!metadata.types?.others?.output) {
```

Would be better to throw an error there?